R/jhu_data.R

```diff
@@ -68,102 +73,176 @@ load_jhu_data <- function(
       ) %>%
       dplyr::select(location, date, cum, inc)
 
     results <- dplyr::bind_rows(results, national_results)
   }
+
   
+   if (replace_negatives == TRUE) {
+    results = covidData::replace_negatives(data = results, measure = measure)
+   }
+
   # TODO: in results data frame, replace daily inc with NA in specific rows, if requested
   # at this point, the results data frame will have daily incidence values and we want to
   # replace the numbers in some rows with NAs (no new rows, editing existing rows)
-  
-  # TODO: if temporal_resolution == 'weekly', aggregate daily incidence to weekly incidence here
+
+  if (adjustment_cases != "none" & length(adjustment_cases) > 0) {
+    adjustment_states <- sub("-.*", "", adjustment_cases)
+    adjustment_dates <- sub("^.*?-", "", adjustment_cases)
+    adjustment_state_fips <- unlist(lapply(
+      adjustment_states, function(x) {
+        covidData::fips_codes[which(covidData::fips_codes$abbreviation == x), ]$location
+      }
+    ))
+    
+
+    adjustments <- data.frame(fips = adjustment_state_fips, dates = as.Date(adjustment_dates))
+    if (adjustment_method == "fill_na") {
+      results <- covidData::fill_na(results = results, adjustments = adjustments)
+    }
+
+    
+    if (adjustment_method == "impute_and_redistribute") {
+       if (replace_negatives == FALSE) {
+         results = covidData::replace_negatives(data = results, measure = measure)
+       }
+
+      for (i in 1:nrow(adjustments)) {
+        adjustment_location <- adjustments[i, ]$fips
+        adjustment_date <- as.Date(adjustments[i, ]$date)
+
+        # Get state, counties and national observations for an adjustment case
+        location_data <- results %>%
+          dplyr::filter(stringr::str_sub(location, start = 1, end = 2) %in% adjustment_location |
+            location == "US" | location == adjustment_location)
+
+        # For each location in data, get imputed data
+        for (fips in unique(location_data$location)) {
+          d <- location_data[location_data$location == fips, ]
+
+          seed <- 1234
+          # Get adjusted inc column
+          imputed_inc <- covidData::adjust_daily_incidence(d, adjustments[i, ], seed, measure = measure)
+
+          # Put imputed data back to results
+          results[which(results$location == fips), ]$inc <- imputed_inc
+        }
+      }
+    }
+  }
+
+
+
+
+  # TODO: if temporal_resolution == "weekly", aggregate daily incidence to weekly incidence here
   # input: data frame with daily values, output: data frame with approximately 1/7 the # of rows
   # with weekly inc values
   # group_by(location, week), and summarize(inc = sum(inc, na.rm = TRUE))
-  
+  # if weekly temporal resolution, filter to saturdays
+  # TODO: adjust data first, go to weekly resolution later
+  if (temporal_resolution == "weekly") {
+    results <- results %>%
+      dplyr::mutate(sat_date = lubridate::ceiling_date
+      (lubridate::ymd(date), unit = "week") - 1) %>%
```

Comment:
For readability, it would be helpful to put the paren starting the function call immediately after the function name, then have the line break.

```suggestion
      dplyr::mutate(sat_date = lubridate::ceiling_date(
        lubridate::ymd(date), unit = "week") - 1) %>%
```